DragAndMeasureTool.cs

```diff
@@ -0,0 +1,339 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public class DragAndMeasureTool : MonoBehaviour, IBeginDragHandler, IDragHandler, IEndDragHandler

```

Comment:
this class feels like it is violating SRP - it seems to do a lot. I think I'd probably break the data transform logic out into separate classes and probably make the display of any text into another class as well